commcare/src/main/java/org/motechproject/commcare/tasks/builder/model/CaseTypeWithDisplayName.java

```diff
@@ -0,0 +1,45 @@
+package org.motechproject.commcare.tasks.builder.model;
+
+/**
+* Stores casetype and it's application name.
+ */
+
+public class CaseTypeWithDisplayName {
+    private String caseType;
+    private String applicationName;
+
+    public CaseTypeWithDisplayName(String caseType, String applicationName) {
+        this.caseType = caseType;
+        this.applicationName = applicationName;
+    }
+
+    public String getCaseType() {
+        return caseType;
+    }
+
+    public void setCaseType(String caseType) {
+        this.caseType = caseType;
+    }
+
+    public String getApplicationName() {
+        return applicationName;
+    }
+
+    public void setApplicationName(String applicationName) {
+        this.applicationName = applicationName;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (caseType.equals((String) obj)) {
```

Comment:
If obj is a String it should never be considered equal to the object of this class.